Expose virtual size of VM images

tserong opened the pull request:
This requires:

- https://github.com/longhorn/backing-image-manager/pull/208
- https://github.com/longhorn/longhorn-manager/pull/2680

With this change, we can do this:

```
# kubectl get vmimages
NAME          DISPLAY-NAME                                         SIZE          VIRTUALSIZE   AGE
image-26r4s   sles15-sp5-minimal-vm.x86_64-kvm-and-xen-qu1.qcow2   267845632     25769803776   6m36s
image-bgnhb   sle-micro.x86_64-5.5.0-default-qcow-gm.qcow2         1001652224    21474836480   20m
image-nmmvj   sle-15-sp5-full-x86_64-gm-media1.iso                 14548992000   14548992000   19m
```

Related issue: https://github.com/harvester/harvester/issues/4905

tserong committed:
Expose virtual size of VM images

With this change, we can do this:

  # kubectl get vmimages
  NAME          DISPLAY-NAME                                         SIZE          VIRTUALSIZE   AGE
  image-26r4s   sles15-sp5-minimal-vm.x86_64-kvm-and-xen-qu1.qcow2   267845632     25769803776   6m36s
  image-bgnhb   sle-micro.x86_64-5.5.0-default-qcow-gm.qcow2         1001652224    21474836480   20m
  image-nmmvj   sle-15-sp5-full-x86_64-gm-media1.iso                 14548992000   14548992000   19m

This in turn means we can update the UI to take image virtual size into
account when creating VMs.

Note that this requires the following Longhorn PRs, which have been
backported to longhorn 1.5.5 and 1.6.2:

- https://github.com/longhorn/backing-image-manager/pull/208
- https://github.com/longhorn/longhorn-manager/pull/2680
- https://github.com/longhorn/longhorn/pull/8267

Related issue: https://github.com/harvester/harvester/issues/4905

Signed-off-by: Tim Serong <tserong@suse.com>

m-ildefons reviewed (changes requested):
I'm unsure about the behavior of this when upgrading from older Harvester versions. The existing (already initialized) images should have their virtual size updated as well. Otherwise images that are grandfathered in from before the upgrade of Harvester may result in VM disks with size 0 being created erroneously.

tserong committed:
Pick up virtual size of existing images after upgrade

Signed-off-by: Tim Serong <tserong@suse.com>

tserong committed:
Ignore BackingImageStateReadyForTransfer when marking vm images imported

When a backing image is in state BackingImageStateReadyForTransfer, it
won't have virtual size yet.  This only happens once it's finally in
BackingImageStateReady, so just wait for that state before marking the
image imported, to keep the code simple.

Signed-off-by: Tim Serong <tserong@suse.com>

bk201 reviewed (approved):
LGTM, thanks.

WebberHuang1118 reviewed (approved):
LGTM, thanks for your effort @tserong .
